pallets/fragments/src/lib.rs

```diff
@@ -329,9 +330,8 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/// Fragment upload function.
-		// TODO #1 - weight
-		#[pallet::weight(25_000)]
+		/// Fragment update function.
+		#[pallet::weight(T::WeightInfo::update())]
```

Comment:
This should be `mutable_data.len()` like `upload`

Reply:
Wrong, as I asked many times we need to take into account the actual data length!